src/transaction.rs

```diff
@@ -31,7 +31,7 @@ where
 
     /// Commit the current transaction changes
     pub fn commit(mut self) -> Result<(), FbError> {
-        self.data.commit_retaining(self.conn)
+        self.data.commit(self.conn)
```

Comment:
We can use `ManuallyDrop` ([link]) to avoid executing the `rollback` in the drop code, as that was the reason i've used
the `commit_retaining` here. What do you think?

[link]: https://doc.rust-lang.org/std/mem/struct.ManuallyDrop.html
```suggestion
        self.data.commit(self.conn)?;
        ManuallyDrop::new(self);
        Ok(())
```

Reply:
Good suggestion

Reply:
Good idea.
1. On failure we still call `drop`, right? Is it intentional to swallow a potential rollback error in this case? (like if disconnected) My understanding is that `isc_rollback_transaction` doesn't really say much on error so it's better to return the commit error if we have to choose here.
2. Is there any reason not to do something similar for `Transaction::rollback`?
3. If we also do it for `Transaction::rollback`, do you have any opinion about calling that explicitly here on failure instead of `TransactionData::rollback` (via `drop`)? Personally I would prefer `Transaction::rollback` since it's local to the impl.

Reply:
1. Yes, as the `ManuallyDrop` is only called after the commit returns successfully.
2. I think it is a good idea to do it in the `rollback` too
3. I don't see any problem in calling it explicitly, and we can use the `ManuallyDrop` in the end of the failure path to avoid `rollback` being called twice.